### ISSUE TITLE
Ignore fields for JSON generation

### ISSUE THREAD

**Issue by gregwebs** [yesodweb/persistent#192]
https://github.com/yesodweb/yesod/issues/571

**Comment by ecognium** [comment]
I came here to add this as an enhancement as I was creating my own FromJSON and ToJSON because I did not want all the fields in my response or incoming requests. I am a total beginner in Haskell so not sure if what I did will work in the general setting. ToJson was the easiest as you pointed out. 

For FromJSON, I just used default pure values. For example, if I had something like: 
```
   User
      name     Text
      email    Text
      verified Bool
```

Even though frontend app should not set whether an account is verified or not, it will be easier if the incoming JSON was of type `User` so it can be inserted in the db quickly (as opposed to another type with `verified` dropped + conversion to the `User` type). The solution I used was to assign default values (say `pure (False)` for verified in  my `toJSON` function).  

In my handler code, I just say something like:
     `runDb $ insert user {verified = False} -- I know it is already set to False but for other derived fields this is a quick way to set the values without additional wrappers.`

Since the application is responsible for ensuring the right data got into the DB anyways (newtype conversion or reset defaults if needed), I felt this is an acceptable compromise. 

May be this will make the DSL too complicated but specifying default value could help alleviate some of the issues (`json(False)`). Although it may not be easy to support complex derived types?
